pipeline/backend/kubernetes/kubernetes.go

```diff
@@ -231,43 +233,34 @@ func (e *kube) WaitStep(ctx context.Context, step *types.Step, taskUUID string)
 			return
 		}
 
-		if pod.Name == podName {
-			if isImagePullBackOffState(pod) {
-				finished <- true
-			}
-			if isCompleted(pod) {
-				finished <- true
-			}
+		if pod.Name != podName {
+			return
+		}
 
-			switch pod.Status.Phase {
-			case v1.PodSucceeded, v1.PodFailed, v1.PodUnknown:
-				finished <- true
-			}
+		if isImagePullBackOffState(pod) || isCompleted(pod) || pod.Status.Phase == v1.PodSucceeded || pod.Status.Phase == v1.PodFailed || pod.Status.Phase == v1.PodUnknown {
+			finished <- true
```

Comment:
Are these changes from line 236 - 241 necessary? I feel they don't really introduce any change, they only reorganize the code.

Reply:
I think this is fine for testing, but I would remove this changes so we don't have to come back and undo them later.